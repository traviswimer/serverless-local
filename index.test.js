const Serverless = require('serverless');
const Promise = require('bluebird');

const ServerlessLocalPlugin = require('./index');

var serverless;
describe('ServerlessLocalPlugin', () => {
	beforeEach(() => {
		serverless = {
			cli: {
				log: jest.fn(),
				consoleLog: jest.fn()
			},
			pluginManager: {
				spawnzz: jest.fn()
			},
			service: {
				resources: {}
			},
			getProvider: jest.fn(() => {
				return {
					sdk: {
						config: {
							update: jest.fn(),
							setPromisesDependency: jest.fn()
						}
					}
				}
			})
		}
	});

	describe('constructor()', () => {
		test('default config is used if no user config provided', () => {
			let plugin = new ServerlessLocalPlugin(serverless);
			expect(plugin.config).toMatchSnapshot();
		});

		test('user config will override default', () => {
			serverless.service.custom = {
				'serverless-local': {
					host: 'http://otherhost',
					region: 'elsewhere',
					ports: {
						"dynamodb": 1111,
						"redshift": 2222,
						"cloudformation": 3333,
						"sqs": 4444,
					},
					endpoints: {
						dynamodbstreams: 'https://randomhost:1234',
						route53: 'http://randomhost:4321'
					}
				}
			}
			let plugin = new ServerlessLocalPlugin(serverless);
			expect(plugin.config).toMatchSnapshot();
		});
	});
});
