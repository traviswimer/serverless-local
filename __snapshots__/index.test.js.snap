// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerlessLocalPlugin constructor() default config is used if no user config provided 1`] = `
Object {
  "endpoints": Object {
    "cloudformation": Object {
      "endpoint": "http://localhost:4581",
    },
    "cloudwatch": Object {
      "endpoint": "http://localhost:4582",
    },
    "dynamodb": Object {
      "endpoint": "http://localhost:4569",
    },
    "dynamodbstreams": Object {
      "endpoint": "http://localhost:4570",
    },
    "es": Object {
      "endpoint": "http://localhost:4578",
    },
    "firehose": Object {
      "endpoint": "http://localhost:4573",
    },
    "kinesis": Object {
      "endpoint": "http://localhost:4568",
    },
    "redshift": Object {
      "endpoint": "http://localhost:4577",
    },
    "route53": Object {
      "endpoint": "http://localhost:4580",
    },
    "s3": Object {
      "endpoint": "http://localhost:4572",
    },
    "ses": Object {
      "endpoint": "http://localhost:4579",
    },
    "sns": Object {
      "endpoint": "http://localhost:4575",
    },
    "sqs": Object {
      "endpoint": "http://localhost:4576",
    },
    "ssm": Object {
      "endpoint": "http://localhost:4583",
    },
  },
  "host": "http://localhost",
  "region": "local",
}
`;

exports[`ServerlessLocalPlugin constructor() user config will override default 1`] = `
Object {
  "endpoints": Object {
    "cloudformation": Object {
      "endpoint": "http://otherhost:3333",
    },
    "cloudwatch": Object {
      "endpoint": "http://otherhost:4582",
    },
    "dynamodb": Object {
      "endpoint": "http://otherhost:1111",
    },
    "dynamodbstreams": Object {
      "endpoint": "https://randomhost:1234",
    },
    "es": Object {
      "endpoint": "http://otherhost:4578",
    },
    "firehose": Object {
      "endpoint": "http://otherhost:4573",
    },
    "kinesis": Object {
      "endpoint": "http://otherhost:4568",
    },
    "redshift": Object {
      "endpoint": "http://otherhost:2222",
    },
    "route53": Object {
      "endpoint": "http://randomhost:4321",
    },
    "s3": Object {
      "endpoint": "http://otherhost:4572",
    },
    "ses": Object {
      "endpoint": "http://otherhost:4579",
    },
    "sns": Object {
      "endpoint": "http://otherhost:4575",
    },
    "sqs": Object {
      "endpoint": "http://otherhost:4444",
    },
    "ssm": Object {
      "endpoint": "http://otherhost:4583",
    },
  },
  "host": "http://otherhost",
  "region": "elsewhere",
}
`;

exports[`ServerlessLocalPlugin portsHandler() updates AWS SDK with config endpoints 1`] = `
Object {
  "cloudformation": Object {
    "endpoint": "http://localhost:4581",
  },
  "cloudwatch": Object {
    "endpoint": "http://localhost:4582",
  },
  "dynamodb": Object {
    "endpoint": "http://localhost:4569",
  },
  "dynamodbstreams": Object {
    "endpoint": "http://localhost:4570",
  },
  "es": Object {
    "endpoint": "http://localhost:4578",
  },
  "firehose": Object {
    "endpoint": "http://localhost:4573",
  },
  "kinesis": Object {
    "endpoint": "http://localhost:4568",
  },
  "redshift": Object {
    "endpoint": "http://localhost:4577",
  },
  "route53": Object {
    "endpoint": "http://localhost:4580",
  },
  "s3": Object {
    "endpoint": "http://localhost:4572",
  },
  "ses": Object {
    "endpoint": "http://localhost:4579",
  },
  "sns": Object {
    "endpoint": "http://localhost:4575",
  },
  "sqs": Object {
    "endpoint": "http://localhost:4576",
  },
  "ssm": Object {
    "endpoint": "http://localhost:4583",
  },
}
`;
